### PR TITLE
Add pre-commit, clang-format and license to files

### PR DESCRIPTION
**Pull request by Kotochleb**
Adds pre-commit configuration to the repository

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,93 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.5.0
+    hooks:
+      - id: check-added-large-files
```
Comment (macstepien) [inline .pre-commit-config.yaml:5]: Does it work with defaults? I checked and max size should be 500kB, and our largest mesh has 1.4MB.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      - id: check-xml
+      - id: check-yaml
+      - id: debug-statements
+      - id: destroyed-symlinks
+      - id: detect-private-key
+      - id: end-of-file-fixer
+      - id: fix-byte-order-marker
+      - id: name-tests-test
+      - id: mixed-line-ending
+      - id: sort-simple-yaml
```
Comment (macstepien) [inline .pre-commit-config.yaml]: I'm not sure about `sort-simple-yaml`

File: panther_battery/launch/battery.launch.py [hunk 3]
```diff
@@ -6,21 +22,21 @@
 
 
 def generate_launch_description():
-    panther_version = LaunchConfiguration('panther_version')
-    declare_panther_version_arg = DeclareLaunchArgument('panther_version')
+    panther_version = LaunchConfiguration("panther_version")
```
Comment (macstepien) [inline panther_battery/launch/battery.launch.py:25]: I thought that we were supposed to change this setting from black and use single quotes (I prefer double though).

File: .pre-commit-config.yaml (hunk 1 above)
Comment (Kotochleb) [reply .pre-commit-config.yaml:5]: I made it to 2 MB

File: .pre-commit-config.yaml (hunk 2 above)
Comment (Kotochleb) [reply .pre-commit-config.yaml]: I removed it

File: panther_battery/launch/battery.launch.py (hunk 3 above)
Comment (Kotochleb) [reply panther_battery/launch/battery.launch.py:25]: As we discussed in ROS 2 we will used double quotes

**Review by macstepien — APPROVED** [review]

**Pull request merged**